### DISCUSSION TITLE
"Chrome is being controlled by automated test software" notification

### DISCUSSION THREAD

**Discussion by GCuser99** [GCuser99/SeleniumVBA#23] [Ideas]
For Chrome and Edge WebDrivers, the default is to get the "Chrome/Edge is being controlled by automated test software" notification near the top of the browser. This does not happen with Firefox WebDriver.

The notification can be disabled using:
```
caps.SetOption "excludeSwitches", Array("enable-automation")
```
The question is, should setting the option above be set as the default in SeleniumVBA? Would there ever be a situation where this notification is useful?

EDIT: ...Upon further testing, the exclude switch above works fine for me in Chrome, but in Edge it causes a "Sync your Profile" popup to occur. I have yet to find a clean solution to avoid the popup in Edge. The only thing that I tried that seems to work was to add the argument "--inprivate", but not sure that running in InPrivate mode is what we want SeleniumVBA to default to.

**Comment by 6DiegoDiego9** [comment]
I confirm the "Sync your profile" issue in Edge.

My preference about the default header, if that issue is solved, is currently about 50/50.

My main pros for keeping it:
- since I always have the same type of browser (Edge) open for my human browsing, it is an immediate distinction mark between the automated and human sessions
- it may be liked by colleagues who use my macros and can help to remark them that there's an automation in progress on the page (so don't interfere with it!)

My main cons:
- 1cm less viewing area

When I'm 50/50 I usually choose the easiest/safest route that is to leave it as it is, until I'll have new pros/cons to consider in future.
What about adding that line and leaving it commented out, in the meantime?

My preference about InPrivate is to not set it as default.  I tend to often want more the opposite, that is to make it use my personal profile like if it were a new window of my human browser session.

**Comment by GCuser99** [comment]
Good feedback - thanks.

I'm refactoring the WebCapabilities class right now (it has not received a lot of attention in the past - it was written in pieces by multiple contributors and so lacks some consistency). I may add a method for "slurping up" a default capabilities json string from file - that way user can optionally and easily customize capabilities to whatever their usual preferences are.
